### PR TITLE
[adaptivecards-templating] Enabled the use of custom functions from adaptive-expressions

### LINKED ISSUE #6133: [Feature Request][PR attached] Enable the use of custom functions within adaptivecards-templating
# Platform

**JavaScript**

# Version of SDK

**adaptivecards-templating**:  
Latest (Present in both NPM latest 2.1.0 and the main branch 2.2.0-beta.1)

# What is the problem you are attempting to solve with this feature?
Currently users are unable to add custom functions to adaptive-expressions and have them parsed in adaptivecards-templating due to the JS library of only searching in the standardFunctions map.

I have created a PR #6132 to enable the use of adaptive-expression custom functions with the JS adaptive-templating package.

# A description of your proposed solution to the problem

The current version of adaptive expressions (and the targeted version in peerDependencies) supports custom functions for templating, with information for the C# version of the library documented [here](https://docs.microsoft.com/en-gb/adaptive-cards/templating/sdk#custom-functions).

This can be supported in the JS version of adaptive-templates by switching out the lookup in the *parseInterpolatedString* method to use the *FunctionTable* of adaptive-expressions, rather than the *standardFunctions* Map, which currently limits developers to only using the standard library.

### PR DESCRIPTION
**Pull request by jsonsCode**
# Related Issue

Fixes #6133

# Description

Custom functions are not currently supported because *parseInterpolatedString* method only looks in the standardFunctions Map for a matching function expression. This PR alters that method to use the *FunctionTable* of adaptive-expressions so that users are able to define their own custom functions.

# How this was verified

1. Ran variations of the sample added to the README in this PR to ensure the functionality works as expected.

2. Tested some of the existing library of standard functions and compared the output of the standardFunctions map to that of the FunctionTable to ensure the existing functionality is preserved.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6132)

### CODE REVIEW

**Comment by dclaux** [comment]
Somehow I had completely missed this. Great add to `adaptivecards-templating`, thanks! @jonmill we should merge this.

File: source/nodejs/adaptivecards-templating/src/template-engine.ts [hunk 1]
```diff
@@ -215,10 +215,10 @@ export class Template {
      */
     public static parseInterpolatedString(interpolatedString: string): AEL.Expression | string {
         let lookup: AEL.EvaluatorLookup = (type: string) => {
-            let standardFunction = AEL.ExpressionFunctions.standardFunctions.get(type);
+            let matchedFunction = AEL.Expression.functions.get(type);
```
Comment (dclaux) [inline source/nodejs/adaptivecards-templating/src/template-engine.ts:218]: Just to be sure as `Expressions.functions` didn't yet exists (as far as I remember) when I coded this originally - the `Expressions.functions` collection combines standard functions and custom ones, correct?

Comment (jsonsCode) [reply source/nodejs/adaptivecards-templating/src/template-engine.ts:218]: Hey,

Thanks for the review, yep that's correct. It references a [FunctionTable instance](https://github.com/microsoft/botbuilder-js/blob/main/libraries/adaptive-expressions/src/functionTable.ts) which combines them both.

**Comment by RebeccaAnne** [comment]
Converting to draft until we have a chance to have a design discussion about this change. (See issue for details)

File: source/nodejs/adaptivecards-templating/README.md [hunk 2]
```diff
@@ -98,7 +98,65 @@ This example is implemented in the **example.html** file.
...
+Custom functions can be added to Adaptive Expression Library in addition to the prebuilt functions.
+Below is an example creating a simple "greet" function that forms a greeting from a user's name.
 
 ```typescript
-// Coming soon...
+const ACData = require("adaptivecards-templating")
+const AdaptiveExpressions = require("adaptive-expressions")
+
+// Register a custom function:
+AdaptiveExpressions.Expression.functions.add("greet", (args) => {
```
Comment (licanhua) [inline source/nodejs/adaptivecards-templating/README.md:109]:
```suggestion
AdaptiveExpressions.Expression.functions.add("sayHello", (args) => {
```
 to match with the example

**Comment by jonmill** [comment]
At this point, I don't think we'll be able to take this PR. I apologize for the lack of traction here...we messed up and didn't commit to making this PR work; that's on me. If you still want this work to be done, please open an issue and we'll start getting traction there to make a path forward.

**Pull request closed without merging**